src/lib/comms/include/sol-gatt.h

```diff
@@ -0,0 +1,256 @@
+				flush, user_data) {			\
+		.type = SOL_GATT_ATTR_TYPE_CHARACTERISTIC,		\
+		.uuid = uuid,						\
+		.flags = flags,						\
+		.read = read,						\
+		.write = write,						\
+		.flush = flush,						\
+		.user_data = user_data,					\
+	}
+
+#define SOL_GATT_DESCRIPTOR(uuid, flags, read, write, flush, user_data) { \
+		.type = SOL_GATT_ATTR_TYPE_DESCRIPTOR,			\
+		.uuid = uuid,						\
+		.flags = flags,						\
+		.read = read,						\
+		.write = write,						\
+		.flush = flush,						\
+		.user_data = user_data,					\
+	}
+
+/**
+ * @brief Registers attributes into the GATT database
+ *
+ * @param attrs Array of the attributes to be registered
+ * @param len Length of the array of attributes
+ *
+ * @return 0 on success, -errno on failure.
+ */
+int sol_gatt_register_attributes(struct sol_gatt_attr *attrs, uint16_t len);
+
+/**
+ * @brief Unregisters attributes from the GATT database
+ *
+ * @param attrs Array of the attributes to be unregistered
+ * @param len Length of the array of attributes
+ *
+ * @return 0 on success, -errno on failure.
+ */
+int sol_gatt_unregister_attributes(struct sol_gatt_attr *attrs, uint16_t len);
+
+/**
+ * @brief Reads the value from a attribute
+ *
+ * This only really makes sense for Characteristics and Descriptors,
+ * the Read Long attribute procedure will be performed if it's supported.
+ *
+ * @param conn Connection to a remote device
+ * @param attr Attribute to be read
+ * @param cb Callback to be called when the operation is finished
+ * @param user_data User pointer to be passed to the callback
+ */
+int sol_gatt_read_attr(struct sol_bt_conn *conn, struct sol_gatt_attr *attr,
+		       void (*cb)(void *user_data, bool success,
+				  struct sol_gatt_attr *attr,
+				  struct sol_buffer *buf),
+		       const void *user_data);
+
+/**
+ * @brief Writes the value to a attribute
+ *
+ * This only really makes sense for Characteristics and Descriptors,
+ * the Write Long attribute procedure will be performed if it's supported.
+ *
+ * @param conn Connection to a remote device
+ * @param attr Attribute to be written
+ * @param cb Callback to be called when the operation is finished
+ * @param user_data User pointer to be passed to the callback
+ */
+int sol_gatt_write_attr(struct sol_bt_conn *conn, struct sol_gatt_attr *attr,
+			struct sol_buffer *buf,
+			void (*cb)(void *user_data, bool success,
+				   struct sol_gatt_attr *attr)
+			const void *user_data);
+
+/**
+ * @brief Discover attributes by type, restricted by UUID and parent attribute.
+ *
+ * Discover attributes belonging to a parent attribute, for example,
+ * discover all the characteristics under a service or all the
+ * descriptors under a characteristic, matching uuid (if provided).
+ */
+int sol_gatt_discover(struct sol_bt_conn *conn, enum sol_gatt_attr_type type,
+		      const struct sol_gatt_attr *parent,
+		      const struct sol_bt_uuid *uuid,
+		      bool (*func)(void *user_data, struct sol_bt_conn *conn,
```

Comment:
s/func/cb/g to be consistent.

also improve the documentation, for all parameters, write which one can be NULL and what it means to return true/false from callback.